### PR TITLE
mesa: fix up dependencies

### PR DESCRIPTION
**Pull request by maxim-belkin**
A follow-up to Linuxbrew/homebrew-xorg#351

### CODE REVIEW

File: Formula/mesa.rb [hunk 1]
```diff
@@ -31,18 +31,23 @@ class Mesa < Formula
...
   depends_on "libelf" # radeonsi requires libelf when using llvm
   depends_on "linuxbrew/xorg/libomxil-bellagio"
-  depends_on "linuxbrew/xorg/wayland-protocols" => :recommended
+  depends_on "linuxbrew/xorg/wayland-protocols" => [:recommended, :build]
+  depends_on "linuxbrew/xorg/xproto" => :build
   depends_on "valgrind" => :recommended
   depends_on "linuxbrew/xorg/libglvnd" => :optional
   depends_on "linuxbrew/xorg/libva" => :recommended
-  depends_on "linuxbrew/xorg/libvdpau"
+  depends_on "linuxbrew/xorg/libvdpau" => :build
```
Comment (sjackman) [inline Formula/mesa.rb:42]: Why no run-time linkage, is it header only? Is this dependency required?

Comment (maxim-belkin) [reply Formula/mesa.rb:42]: `libvdpau` is not header-only:
```
/home/linuxbrew/.linuxbrew/Cellar/libvdpau/1.1.1/lib/libvdpau.so
/home/linuxbrew/.linuxbrew/Cellar/libvdpau/1.1.1/lib/vdpau/libvdpau_trace.so
```
But mesa does not link against these. I _think_ `mesa` uses `libvdpau` to build libs like

```
/home/linuxbrew/.linuxbrew/Cellar/mesa/17.2.3_3/lib/vdpau/libvdpau_r300.so
/home/linuxbrew/.linuxbrew/Cellar/mesa/17.2.3_3/lib/vdpau/libvdpau_radeonsi.so
/home/linuxbrew/.linuxbrew/Cellar/mesa/17.2.3_3/lib/vdpau/libvdpau_nouveau.so
```

Comment (maxim-belkin) [reply Formula/mesa.rb:42]: > Is this dependency required?

yes, it is required - the build fails without it.

Comment (maxim-belkin) [reply Formula/mesa.rb:42]: I see several 
```
-I/home/linuxbrew/.linuxbrew/Cellar/libvdpau/1.1.1/include
```
in `mesa` build logs... perhaps, mesa uses header files only, actually...

Comment (sjackman) [reply Formula/mesa.rb:42]: 👍 That's possible.

**Comment by sjackman** [comment]
Thanks, Maxim!

**Pull request merged**